cdap-ui/app/cdap/components/NamespaceDetails/Preferences/index.js

```diff
@@ -41,130 +46,205 @@ const PREFERENCES_TABLE_HEADERS = [
+      }, 3000);
+    });
+  }
+
+  componentDidUpdate() {
+    let highlightedElems = document.getElementsByClassName('highlighted');
+    if (highlightedElems.length) {
+      highlightedElems[0].scrollIntoView();
+    }
+  }
+
+  componentWillUnmount() {
+    this.eventEmitter.off(globalEvents.NSPREFERENCESSAVED, getNamespacePrefs);
+  }
+
+  toggleModal = () => {
+    this.setState({
+      modalOpen: !this.state.modalOpen
+    });
   };
-};
 
-const PreferencesTable = ({namespacePrefs, systemPrefs}) => {
-  if (isEmpty(namespacePrefs) && isEmpty(systemPrefs)) {
+  toggleViewAll = () => {
+    this.setState({
+      viewAll: !this.state.viewAll
+    });
+  }
+
+  convertPrefsArray(prefs, scope) {
+    let sortedPrefObjectKeys = [...Object.keys(prefs)].sort();
+    return sortedPrefObjectKeys.map(prefKey => {
+      return {
+        key: prefKey,
+        scope,
+        value: prefs[prefKey],
+        highlighted: this.state.newNamespacePrefsKeys.indexOf(prefKey) !== -1
+      };
+    });
+  }
+
+  renderPreferencesLabel() {
+    let label;
+    if (isEmpty(this.state.namespacePrefs) && isEmpty(this.state.systemPrefs)) {
+      label = <strong>{T.translate(`${PREFIX}.label`)}</strong>;
+    } else {
+      let prefsCount = Object.keys(this.state.namespacePrefs).length + Object.keys(this.state.systemPrefs).length;
+      label = <strong>{T.translate(`${PREFIX}.labelWithCount`, {count: prefsCount})}</strong>;
+    }
+
     return (
-      <div className="text-xs-center">
-        {T.translate(`${PREFIX}.noPreferences`)}
+      <div className="namespace-details-section-label">
+        {label}
+        <span
+          className="edit-label"
+          onClick={this.toggleModal}
+        >
+          {T.translate('features.NamespaceDetails.edit')}
+        </span>
       </div>
     );
   }
 
-  namespacePrefs = convertPrefsArray(namespacePrefs, T.translate('features.NamespaceDetails.namespace'));
-  systemPrefs = convertPrefsArray(systemPrefs, T.translate('commons.cdap'));
-  let prefs = namespacePrefs.concat(systemPrefs);
-
-  return (
-    <SortableTable
-      entities={prefs}
-      tableHeaders={PREFERENCES_TABLE_HEADERS}
-      renderTableBody={renderPreferencesTableBody}
-      className="preferences-table"
-      sortOnInitialLoad={false}
-    />
-  );
-};
+  renderPreferencesTable() {
+    if (isEmpty(this.state.namespacePrefs) && isEmpty(this.state.systemPrefs)) {
+      return (
+        <div className="text-xs-center">
+          {T.translate(`${PREFIX}.noPreferences`)}
+        </div>
+      );
+    }
 
-PreferencesTable.propTypes = {
-  namespacePrefs: PropTypes.object,
-  systemPrefs: PropTypes.object
-};
+    let namespacePrefs = this.convertPrefsArray(this.state.namespacePrefs, T.translate('features.NamespaceDetails.namespace'));
+    let systemPrefs = this.convertPrefsArray(this.state.systemPrefs, T.translate('commons.cdap'));
+    let prefs = namespacePrefs.concat(systemPrefs);
+    prefs = orderBy(prefs, ['key']);
```

Comment:
I think this ordering function should be handled when you get the data. This orderBy will execute every time on render, but the data itself might not have changed